### ISSUE TITLE
Removing big files from a repository's history

### ISSUE THREAD

**Issue by avdeev** [shower/shower#324]
To create a new presentation, we need to clone the repository, which size about 17 MiB. Can we delete large files from history to make this process faster?

```bash
Mac-mini-Alexey:shower alexeyavdeev$ ./git-show-big-files.sh
All sizes are in kB's. The pack column is the size of the object, compressed, inside the pack file.
size  SHA                                       location
5410  4c032ce66fcd2ade78e5c2929ab7d3a37bf2f241  archive.zip
2139  855b89cdd30ee8ac4ebb374233d5c67bad712b16  archive.zip
627   9d7d03b7df5035c02709b3b818096aa9d140251c  fonts/PTSans.ttf
585   684526070b0e5f08cc97bff5ae5d8bfe821e1fb6  fonts/PTSans.Italic.ttf
567   3cef37f5e4dee8ab1f4fdc98a8932ee080a0c0af  fonts/PTSans.Narrow.ttf
529   3d66e3289f541acf05980bb9860dd1c27b91cd1c  fonts/PTSans.Caption.ttf
464   cbc21827a773aa1d06c0b37e06adc6fc2a09b522  themes/ribbon/fonts/PTSans.Narrow.Bold.ttf
447   4f038bdcee60f87c1d6eddec9af6c90df72ac1f3  themes/ribbon/fonts/PTSans.Bold.ttf
420   cc0d9088aa4e06b7103511d4d86a488ad5248bae  themes/ribbon/fonts/PTSans.ttf
```

**Comment by pepelsbey** [comment]
If you know how to clean it up, please go ahead and create PR. But it might break some forks.

**Comment by avdeev** [comment]
Yes you are right. Removing files from the history is an unsafe operation.

We can use shallow cloning instead.
https://git-scm.com/docs/git-clone#git-clone---depthltdepthgt

**Comment by avdeev** [comment]
**708.43 KiB** vs **17.79 MiB**

```sh
Mac-mini-Alexey:projects alexeyavdeev$ git clone --depth=1 git@github.com:shower/shower.git
Cloning into 'shower'...
remote: Counting objects: 29, done.
remote: Compressing objects: 100% (27/27), done.
remote: Total 29 (delta 0), reused 18 (delta 0), pack-reused 0
Receiving objects: 100% (29/29), 708.43 KiB | 584.00 KiB/s, done.
```

```sh
Mac-mini-Alexey:projects alexeyavdeev$ git clone git@github.com:shower/shower.git shower-1
Cloning into 'shower-1'...
remote: Counting objects: 2632, done.
remote: Compressing objects: 100% (15/15), done.
remote: Total 2632 (delta 5), reused 7 (delta 2), pack-reused 2615
Receiving objects: 100% (2632/2632), 17.79 MiB | 1.32 MiB/s, done.
Resolving deltas: 100% (1307/1307), done.
```

**Comment by pepelsbey** [comment]
Somewhat fixed in #325
